Removed meteo extractor

GriffinBabe opened the pull request:
@HansVRP I removed the METEO fetcher as the same functionalities are present in the METEO fetcher

VincentVerelst reviewed (commented):
One question: this does mean we don't support fetching meteo from the openeo STAC API anymore. Should we then maybe add an assertion that if "AGERA5" is selected, only the terrascope backend can be used?

GriffinBabe commented:
@VincentVerelst I added the functionality `build_generic_extractor_stac` to continue supporting stac extraction, and added some checks based on the backend.

From a discussion with @HansVRP we decided that GFMAP shouldn't take too much care of checking if a collection is valid or not in a certain backend but rather instead let the `load_collection` method raise the exception itself.

GriffinBabe commented:
@VincentVerelst can you review this PR ?

VincentVerelst commented on src/openeo_gfmap/fetching/generic.py:
```diff
@@ -28,18 +28,36 @@
         band_mapping = BASE_WEATHER_MAPPING
+    elif is_stac and (AGERA5_TERRASCOPE_STAC in collection_name):
+        if backend not in [Backend.TERRASCOPE, Backend.FED]:
+            raise ValueError(
```
Should this error not be thrown if `is_stac==False` and the backend is not Terrascope or Fed?

GriffinBabe replied:
Yes you are right, I removed the check. Actually:

> From a discussion with @HansVRP we decided that GFMAP shouldn't take too much care of checking if a collection is valid or not in a certain backend but rather instead let the load_collection method raise the exception itself.
